test/extensions/implementation/autograd.py

```diff
@@ -29,3 +29,10 @@ def batch_grad(self):
                 batch_grads[idx][b, :] = g.detach() * factor
 
         return batch_grads
+
+    def batch_l2_grad(self):
+        batch_grad = self.batch_grad()
+        batch_l2_grads = [
+            (g ** 2).sum(list(range(1, len(g.shape)))) for g in batch_grad
```

Comment:
The summation over all axes except the `0`th could also be written as `(g ** 2).flatten(start_dim=1).sum(1) for g in ...`. Do you think that this way the code would be easier to read?

Reply:
Yes, this is a better way to write it